--- src/parsing/color_parsing_helper.c
+++ src/parsing/color_parsing_helper.c
@@ -16,15 +16,18 @@
 inteparse_floor_color(t_map *map, const char *line, int *parsed)
 {
 	inteout[3];
+vi-intres;
 
- if (!parse_color(line + 1, out))
- return (0);
+vi-res = parse_color(line + 1, out);
+vi-if (res <= 0)
+vi-return (res);
 	map->floor_rgb[0] = out[0];
 	map->floor_rgb[1] = out[1];
 	map->floor_rgb[2] = out[2];
 	(*parsed)++;
 	return (1);
 }
+
 /*
 ** parse_ceiling_color: Parses C ceiling color (R,G,B).
 */
@@ -32,15 +35,18 @@
 inteparse_ceiling_color(t_map *map, const char *line, int *parsed)
 {
 	inteout[3];
+vi-intres;
 
- if (!parse_color(line + 1, out))
- return (0);
+vi-res = parse_color(line + 1, out);
+vi-if (res <= 0)
+vi-return (res);
 	map->ceil_rgb[0] = out[0];
 	map->ceil_rgb[1] = out[1];
 	map->ceil_rgb[2] = out[2];
 	(*parsed)++;
 	return (1);
 }
